api/submission/fetch_submission_history_test.go

```diff
@@ -0,0 +1,122 @@
+        {model.RoleStudent, "grader@test.com", true, false, true, nil},
+
+        // Owner, self.
+        {model.RoleOwner,   "",                true, false, false, nil},
+        {model.RoleOwner,   "owner@test.com",  true, false, false, nil},
+
+        // Owner, other.
+        {model.RoleOwner,   "student@test.com",true, true, false, studentGradingResults},
+
+        // Owner, missing.
+        {model.RoleOwner,   "ZZZ@test.com",    false, false, false, nil},
+
+        // Admin, self.
+        {model.RoleAdmin,   "",                true, false, false, nil},
+        {model.RoleAdmin,   "owner@test.com",  true, false, false, nil},
+
+        // Admin, other.
+        {model.RoleAdmin,   "student@test.com",true, true, false, studentGradingResults},
+
+        // Admin, missing.
+        {model.RoleAdmin,   "ZZZ@test.com",    false, false, false, nil},
+
+    };
+
+    for i, testCase := range testCases {
+        field := map[string]any{
+            "target-email": testCase.targetEmail,
+        };
+
+        response := core.SendTestAPIRequestFull(test, core.NewEndpoint(`submission/fetch/history`), field, nil, testCase.role);
+        if (!response.Success) {
+            if (testCase.permError) {
+                expectedLocator := "-020";
+                if (response.Locator != expectedLocator) {
+                    test.Errorf("Case %d: Incorrect error returned on permissions error. Expcted '%s', found '%s'.",
+                        i, expectedLocator, response.Locator);
+                }
+            } else {
+                test.Errorf("Case %d: Response is not a success when it should be: '%v'.", i, response);
+            }
+
+            continue;
+        }
+
+        var responseContent FetchSubmissionHistoryResponse;
+        util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent);
+
+        if (testCase.foundUser != responseContent.FoundUser) {
+            test.Errorf("Case %d: Found user does not match. Expected: '%v', actual: '%v'.", i, testCase.foundUser, responseContent.FoundUser);
+            continue;
+        }
+
+        if (testCase.foundSubmissions != responseContent.FoundSubmissions) {
+            test.Errorf("Case %d: Found submissions does not match. Expected: '%v', actual: '%v'.", i, testCase.foundSubmissions, responseContent.FoundSubmissions);
+            test.Errorf("Expected: %v, actual: %v", testCase.result, responseContent.GradingResults)
+            continue;
+        }
+
+        if (!responseContent.FoundSubmissions) {
+            continue;
+        }
+
+        if (!reflect.DeepEqual(testCase.result, responseContent.GradingResults)) {
+            test.Errorf("Case %d: Unexpected submission result. Expected: '%s', actual: '%s'.", i,
+                util.MustToJSONIndent(testCase.result), util.MustToJSONIndent(responseContent.GradingResults));
+            continue;
+        }
+    }
+}
+
+func getTestSubmissionsResultPath(shortID string) string {
```

Comment:
Just use the existing function from fetch_submission_test.go anything in the package can use unexported functions.